ReactiveCocoa/Swift/Property.swift

```diff
@@ -63,18 +63,29 @@ public final class MutableProperty<T>: MutablePropertyType {
 
 	private let observer: Signal<T, NoError>.Observer
 
+	/// Need a recursive lock around `value` to allow recursive access to
+	/// `value`. Note that recursive sets will still deadlock because the
+	/// underlying producer prevents sending recursive events.
+	private let lock = NSRecursiveLock()
```

Comment:
Can you set `lock.name` in the initializer so deadlocks describe the source more clearly?